examples/ecs/state.rs

```diff
@@ -18,7 +18,8 @@ fn main() {
         .add_systems(OnEnter(AppState::Menu), setup_menu)
         // By contrast, update systems are stored in the `Update` schedule. They simply
         // check the value of the `State<T>` resource to see if they should run each frame.
-        .add_systems(Update, menu.run_if(in_state(AppState::Menu)))
+        .add_systems(Update, menu_interaction.run_if(in_state(AppState::Menu)))
+        .add_systems(Update, menu_action.run_if(on_event::<Click>()))
```

Comment:
One last suggestion. Could you group up all systems in this example that run in `Update` into a single call? Just use tuple. I believe that these functions separated now because before we had `OnUpdate(state)`.